x-pack/plugin/core/build.gradle

```diff
@@ -176,3 +176,16 @@ if (BuildParams.inFipsJvm){
   // Test clusters run with security disabled
   tasks.named("javaRestTest").configure{enabled = false }
 }
+
+//this specific test requires a test only system property to be set, so we run it in a different JVM via a separate task
+tasks.register('testAutomatonPatterns', Test) {
+  include '**/AutomatonPatternsTests.class'
+  systemProperty 'tests.automaton.record.patterns', 'true'
```

Comment:
I assume it's not ok to just set this system property for all other tests as well?

Reply:
It probably wouldn't hurt to run all of our tests with this set for all of them, but then we would be running all of our tests in a configuration that does not match how production code is ever run. We could also just change production code to always record the patterns, but it adds a tiny bit of overhead that is only ever used by this one test. The special test task here is a bit awkward, but IMO is better than alternatives.

Reply:
Got it, I wasn't sure if it was completely benign and nonapplicable to the other tests runtime behavior.